src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -287,6 +287,12 @@ protected virtual string GenerateUniqueId()
                         continue;
                     }
 
+                    if (attributeValueChunk is PreallocatedTagHelperAttributeChunk)
+                    {
+                        RenderPreallocatedBoundAttribute(attributeName, attributeValueChunk);
+                        continue;
```

Comment:
This means preallocated bound attributes are added to the `AllAttributes` collection but their values are never assigned to the associated tag helper properties. Suggest this special case needs to live inside `RenderNewAttributeValueAssignment()`.